glib/src/object.rs

```diff
@@ -1838,7 +1870,23 @@ impl<T: ObjectType> ObjectExt for T {
         Ok(())
     }
 
-    fn property<'a, N: Into<&'a str>>(&self, property_name: N) -> Result<Value, BoolError> {
+    fn try_property<'a, N: Into<&'a str>, V: FromValue<'static>>(
+        &self,
+        property_name: N,
+    ) -> Result<V, BoolError> {
+        let prop = self.try_property_value(property_name)?;
+        prop.get::<V>()
```

Comment:
This doesn't work for lifetime reasons 
```

error[E0597]: `prop` does not live long enough
    --> glib/src/object.rs:1878:9
     |
1878 |         prop.get::<V>()
     |         ^^^^-----------
     |         |
     |         borrowed value does not live long enough
     |         argument requires that `prop` is borrowed for `'static`
1879 |             .map_err(|e| crate::bool_error!("Failed to get cast value to a different type {}", e))
```
tried everything i know but failed everytime

Reply:
As discussed on matrix, we should add a `FromValueOwned` trait that is used here and everywhere else where currently a `FromValue<'static>`  is used. Something like

```rust
pub trait FromValueOwned: for<'a> FromValue<'a> {
    unsafe fn from_value_owned(v: &Value) -> Self;
}
```

Reply:
And a `Value::get_owned()` that makes use of this trait

Reply:
> As discussed on matrix, we should add a `FromValueOwned` trait that is used here and everywhere else where currently a `FromValue<'static>`  is used. Something like
> 
> ```rust
> pub trait FromValueOwned: for<'a> FromValue<'a> {
>     unsafe fn from_value_owned(v: &Value) -> Self;
> }
> ```


Should I go ahead and add it as part of the same PR or do it in a different one ?

Reply:
A different one would be best, and then this one can be rebased on top of that once merged.

If you can write a `Value::get_owned()` function then this PR will also be implementable.